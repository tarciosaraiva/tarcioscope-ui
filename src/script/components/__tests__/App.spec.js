import { shallowMount } from '@vue/test-utils'

import App from '../App.vue'
import AppHeader from '../AppHeader'
import AppFooter from '../AppFooter'
import CameraWrapper from '../CameraWrapper'

describe('<App />', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(App, {
      stubs: { 'vue-progress-bar': true }
    })
  })

  it('renders the main block', () => {
    expect(wrapper.contains('main')).toBe(true)
    expect(wrapper.contains(AppHeader)).toBe(true)
    expect(wrapper.contains(AppFooter)).toBe(true)
    expect(wrapper.contains(CameraWrapper)).toBe(true)
  })

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
