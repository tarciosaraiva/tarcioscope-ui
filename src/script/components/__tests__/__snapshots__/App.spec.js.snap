// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<main role="main" class="container">
  <vue-progress-bar-stub></vue-progress-bar-stub>
  <app-header-stub></app-header-stub>
  <camera-wrapper-stub class="camera-wrapper"></camera-wrapper-stub>
  <app-footer-stub></app-footer-stub>
</main>
`;
